src/js/player.js

```diff
@@ -1255,6 +1266,24 @@ class Player extends Component {
     return;
   }
 
+  middlewareIterator(method) {
+    return (value, mw) => {
+      if (mw[method]) {
+        return mw[method](value);
+      }
+
+      return value;
+    };
+  }
+
+  middlewareSet(method, arg) {
+    return this.middleware_.reduce(this.middlewareIterator(method), arg);
+  }
+
+  middlewareGet(method) {
+    return this.middleware_.reduceRight(this.middlewareIterator(method), undefined);
+  }
```

Comment:
It feels like these three middleware functions might belong in the `middleware.js` file instead. Just be that they accept a list of middlewares to process that would be passed in.